pkg/daemon/daemon.go

```diff
@@ -38,6 +38,18 @@ func New(cfg *config.EnvConfig) (srv *Daemon, err error) {
 
 	r := mux.NewRouter()
 
+	// Authentication
+	r.Use(func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
We can initialise a map[string]struct{} inside the closure, and close over it (capture it) in the nested closure. Then we can get rid of the contains method. BTW -- when dealing with critical config values like this, I really recommend cleaning up the user-input strings using strings.TrimSpace.